src/os_posix/os_fs.c

```diff
@@ -733,7 +911,11 @@ __posix_open_file(WT_FILE_SYSTEM *file_system, WT_SESSION *wt_session, const cha
 #ifdef HAVE_FTRUNCATE
     file_handle->fh_truncate = __posix_file_truncate;
 #endif
-    file_handle->fh_write = __posix_file_write;
+
+    if (conn->mmap)
+        file_handle->fh_write = __posix_file_write_mmap;
+    else
+        file_handle->fh_write = __posix_file_write;
```

Comment:
I didn't look at this hard, but since `read/write mmap` has to fall back to a system call, maybe we just write one function?

Reply:
Yes, we could do that.

Reply:
@keithbostic, I've looked at it again. There is no duplication in the __posix_file_read/write_mmap over the non-mmap counterparts. I don't see any downside for keeping the current design with two functions. So I am leaving things as is for now, but if you prefer to not have two functions, that would be an easy change.